io_section.tex

```diff
@@ -0,0 +1,257 @@
+      \hline
+      24 & 83.0 & 81.8 & 77.4 \\
+      54 & 150.6 & 149.2 & 138.8 \\
+      96 & ? & ? & ? \\
+    \end{tabular}
+  \end{center}
+\end{table}
+
+\begin{table}[ht!]
+\scriptsize
+  \begin{center}
+    \caption{Regrid, precomputed weights: Runtime}
+    \label{tab:table7}
+     \begin{tabular}{|c|c|c|c|c|c|}
+      \textbf{Nodes} & \textbf{Max CR~(\%)} & \textbf{Max Client IO~(s)} & \textbf{Runtime~(s)} & \textbf{\% incr diag IO} & \textbf{\% incr no IO}\\
+      \hline
+      24 & 0.02 & 7.07 & 440 & +3 & +30 \\
+      54 & 0.04 & 3.30 & 288 & -18 & -3 \\
+      96 & 0.04 & 4.32 & 307 & -3 & +12 \\
+    \end{tabular}
+  \end{center}
+\end{table}
+
+\begin{table}[ht!]
+\scriptsize
+  \begin{center}
+    \caption{Regrid, precomputed weights: Memory Usage}
+    \label{tab:table8}
+     \begin{tabular}{|c|c|c|c|}
+      \textbf{Nodes} & \textbf{Memory Usage Gb (pp) } & \textbf{Memory Usage Gb (diag only)} & \textbf{Memory Usage Gb (no IO)} \\
+      \hline
+      24 & 85.4 & 81.8 & 77.4 \\
+      54 & 153.0 & 149.2 & 138.8 \\
+      96 & 248.2 & ? & ? \\
+    \end{tabular}
+  \end{center}
+\end{table}
+
+\normalsize
+
+From looking at the comparative runtimes in Figures \ref{fig:fig2} and \ref{fig:fig3} alone there is no clear benefit of computing weights each time vs using precomputed weights. As in previous results the general run time variability obscures things. Scaling is still evident as run times generally decrease with
+increasing node count. Table \ref{tab:table7} gives a slight hint that client performance improves when using precomputed weights (lower CR and time spent in I/O).
+Interestingly, memory usage increases slightly. Presumably there is a trade-off between the time and resources spent reading in the weights vs computing them.
+
+\subsection{Conclusions}
+
+Compared to the previous I/O performance assessment, in this year's tests the mesh resolution has been doubled and many more fields are output (although fewer timesteps are being run). A first assessment of the cost of in-situ post-processing has been made and is generally encouraging in that it seems to work correctly and there is no severe overhead of adding post-processing on top of regular diagnostic output. However, to get a really clear picture a thorough analysis of runtime variability for all non-I/O, diagnostic and post-processing scenarios is necessary. The post-processing tests need to be repeated with a higher resolution mesh as it may make any penalty more obvious (e.g. when computing regridding weights on the fly vs using precomputed weights). The issues seen with higher node count jobs need addressing to find out the cause. In the last I/O performance assessment no problem was seen with 216 and 384 node jobs but many things have changed since then: the mesh is higher resolution and the gungho science has changed considerably. 
```

Comment:
"gungho" -> "GungHo". Also, is it "science scheme" or "dynamical scheme"?

Reply:
dynamics and physics science has changed considerably. Probably better to say 'LFRic dynamics and physics science schemes'